lib/spack/spack/cmd/find.py

```diff
@@ -293,23 +297,27 @@ def root_decorator(spec, string):
         )
         print()
 
-    if args.show_concretized:
+    if args.show_concretized and not args.constraint:
         tty.msg("Concretized roots")
         cmd.display_specs(env.specs_by_hash.values(), args, decorator=decorator)
         print()
 
-    # Display a header for the installed packages section IF there are installed
-    # packages. If there aren't any, we'll just end up printing "0 installed packages"
-    # later.
-    if results and not args.only_roots:
-        tty.msg("Installed packages")
-
 
 def find(parser, args):
-    q_args = query_arguments(args)
-    results = args.specs(**q_args)
-
     env = ev.active_environment()
+
+    if args.show_concretized:
+        if not env:
+            tty.die("-c/specs given but no active environment to search")
```

Comment:
make this consistent with the other error message below.
```suggestion
             tty.die("-c / --show-concretized requires an active environment")
```